spec/iasm.dd

```diff
@@ -26,6 +26,38 @@ $(SPEC_S D x86 Inline Assembler,
         $(D D_InlineAsm_X86_64) version identifiers, respectively.
     )
 
+$(H2 $(LNAME2 asmstatements, Asm statement))
+
+$(GRAMMAR
+$(GNAME AsmStatement):
+    $(D asm) $(GLINK FunctionAttributes)$(OPT) $(D {) $(GLINK AsmInstructionList)$(OPT) $(D })
+
+    $(GNAME AsmInstructionList):
+        $(GLINK AsmInstruction) $(D ;)
+        $(GLINK AsmInstruction) $(D ;) $(I AsmInstructionList)
+    )
+
+    $(P Assembler instructions must be located inside an `asm` block.
+        Like functions, `asm` statements must be anotated with adequate function attributes to be compatible with the caller.
+        Asm statments attributes must be explicit, they are not infered.
+    )
+
+---
+void func1() pure nothrow @safe @nogc
+{
+    asm pure nothrow @trusted @nogc
+    {}
```

Comment:
How about actually using assembler statements here?

Reply:
Assuming you meant instructions and not ~~statements~~.
i don't think that this is required. The scope of the documentation here is the asm statement, not the instructions.